lib/datasource/remote/model/transaction_model.dart

```diff
@@ -25,15 +25,20 @@ class TransactionModel extends Equatable {
   @override
   List<Object?> get props => [transactionId];
 
-  factory TransactionModel.fromJson(Map<String, dynamic> json) {
-    return TransactionModel(
-      from: json['act']['data']['from'],
-      to: json['act']['data']['to'],
-      quantity: json['act']['data']['quantity'],
-      memo: json['act']['data']['memo'],
-      timestamp: parseTimestamp(json['@timestamp']),
-      transactionId: json['trx_id'],
-    );
+  static TransactionModel? fromJson(Map<String, dynamic> json) {
```

Comment:
Some issues here: 

1 - Any "catch" statement should have an error message, we never want an empty catch. It's very confusing. 

2 -Not sure under what conditions a transaction would not have the required fields? Or is that how we filter out transactions? I would prefer a more explicit way in this case... and I think we have some code that does that...

3 - This should be called TransferAction, not "transaction" - it's the wrong name. Although that goes way back, has nothing to do with this PR. Can be done in future PR. 

At minimum, if this is a good use of an exception, log the error.

Reply:
(2-) This fix was triggered by the fact that if there was an atomicassets NFT transfer in a user's transaction history, the app wasn't able to display the history at all. That's because there is no "amount" field in an NFT transfer and the original factory would fail.

(1-) This code adds the semantics that the returned TransactionModel value is nullable, and the null value means "this transfer is not a vanilla token transfer; you may ignore it and carry on".
 There is no need to create an error message.
 
(3-) Agree, I noticed this long ago and kicked the can down the road.

Reply:
(2) I see - this is a good fix. 

(1) perhaps thats the intended semantics but it's not the actual semantics

The actual semantics are: If there is an error thrown here, for any reason whatsoever, ignore it and move on. We never want to do this in the codebase. The reason is, other errors may trigger this, and whoever is reviewing it will spend hours figuring it out - we don't want to use throw/catch in normal conditions, we only want to use it on actual error conditions and in this case it needs to log the error, or even surface it in the UX. 

It's a policy if you will. 

A better test would be - does this transaction have all the fields I need? And then ignore some of them based on that. 
I think I even had an "istransaction" function somewhere, could be used for that.

Reply:
ok I did this but should be separate PR since we need to test, I need to get sent an NFT to my account!

https://github.com/JoinSEEDS/seeds_light_wallet/pull/1956